test_scripts/Defects/7_1/1896_SDL_does_not_send_HeartBeat_request_to_Mobile_HeartBeatTimeout_is_Zero.lua

```diff
@@ -0,0 +1,99 @@
+local events = require('events')
+
+--[[ Test Configuration ]]
+config.defaultProtocolVersion = 3
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local Functions ]]
+local mobSessionConfig = {
+  activateHeartbeat = false,
+  sendHeartbeatToSDL = false,
+  answerHeartbeatFromSDL = true,
+  ignoreSDLHeartBeatACK = true
+}
+
+local function registerApp(pMobSessionConfig)
+  local session = common.mobile.createSession(1, 1, pMobSessionConfig)
+  session:StartService(7)
+  :Do(function()
+      local corId = session:SendRPC("RegisterAppInterface", common.app.getParams(1))
+      common.hmi.getConnection():ExpectNotification("BasicCommunication.OnAppRegistered",
+        { application = { appName = common.app.getParams(1).appName } })
+      :Do(function(_, d1)
+          common.app.setHMIId(d1.params.application.appID, 1)
+        end)
+      session:ExpectResponse(corId, { success = true, resultCode = "SUCCESS" })
+      :Do(function()
+          session:ExpectNotification("OnHMIStatus",
+            { hmiLevel = "NONE", audioStreamingState = "NOT_AUDIBLE", systemContext = "MAIN" })
+          session:ExpectNotification("OnPermissionsChange")
+          :Times(AnyNumber())
+        end)
+    end)
+end
+
+local function WithoutHBFromSDLMsg(pSession)
+  local event = events.Event()
+  event.matches = function(_, data)
+    return data.frameType == constants.FRAME_TYPE.CONTROL_FRAME
+      and data.serviceType == constants.SERVICE_TYPE.CONTROL
+      and data.sessionId == common.getMobileSession(1).sessionId
+      and data.frameInfo == constants.FRAME_INFO.HEARTBEAT
+  end
+  return pSession:ExpectEvent(event, "HB"):Times(0)
+end
+
+local function HBACKFromSDLMsg(pSession)
+  local event = events.Event()
+  event.matches = function(_, data)
+    return data.frameType == constants.FRAME_TYPE.CONTROL_FRAME
+      and data.serviceType == constants.SERVICE_TYPE.CONTROL
+      and data.sessionId == common.getMobileSession(1).sessionId
+      and data.frameInfo == constants.FRAME_INFO.HEARTBEAT_ACK
+  end
+  return pSession:ExpectEvent(event, "HBACK")
+end
+
+local function sendHBFromMobile()
+  local session = common.getMobileSession(1)
+  -- Send HB from mobile app to SDL
+  session:Send({
+    frameType = constants.FRAME_TYPE.CONTROL_FRAME,
+    serviceType = constants.SERVICE_TYPE.CONTROL,
+    frameInfo = constants.FRAME_INFO.HEARTBEAT
+  })
+  -- Expect HB from SDL on mobile app
+  WithoutHBFromSDLMsg(session)
+  -- Expect HB ACK from SDL on mobile app
+  HBACKFromSDLMsg(session)
+end
+
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", common.preconditions)
+runner.Step("Set HeartBeatTimeout=0 in ini file", common.setSDLIniParameter,
+  { "HeartBeatTimeout", 0 })
+runner.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+runner.Step("App is registered", registerApp, { mobSessionConfig })
+runner.Step("Activate App", common.activateApp)
```

Comment:
I believe that app activation can be omitted(remove activation step from the description in case activation will be removed)